qiskit_ibm_experiment/client/local_client.py

```diff
@@ -0,0 +1,534 @@
+
+        Returns:
+            A list of experiments and the marker, if applicable.
+        """
+        df = self._experiments
+
+        if experiment_type is not None:
+            if experiment_type[:5] == 'like:':
+                experiment_type = experiment_type.split(":")[1]
+                df = df.loc[df.type.str.contains(experiment_type)]
+            else:
+                df = df.loc[df.type == experiment_type]
+
+        if backend_name is not None:
+            df = df.loc[df.device_name == backend_name]
+
+        # Note a bug in the interface for all services:
+        # It is impossible to filter by experiments whose parent id is None
+        # (i.e., root experiments)
+        if parent_id is not None:
+            df = df.loc[df.parent_experiment_uuid == parent_id]
+
+        # Waiting for consistency between provider service and qiskit-experiments service,
+        # currently they have different types for `device_components`
+        if device_components is not None:
+            raise ValueError(
+                "The fake service currently does not support filtering on device components"
+            )
+
+        if tags is not None:
+            if tags_operator == "OR":
+                df = df.loc[df.tags.apply(lambda dftags: any(x in dftags for x in tags))]
+            elif tags_operator == "AND":
+                df = df.loc[df.tags.apply(lambda dftags: all(x in dftags for x in tags))]
+            else:
+                raise ValueError("Unrecognized tags operator")
+
+        # These are parameters of IBMExperimentService.experiments
+        if "start_datetime_before" in filters:
+            df = df.loc[df.start_time <= filters["start_datetime_before"]]
+        if "start_datetime_after" in filters:
+            df = df.loc[df.start_time >= filters["start_datetime_after"]]
+
+        # This is a parameter of IBMExperimentService.experiments
+        sort_by = filters.get("sort_by")
+        if sort_by is None:
+            sort_by = "start_datetime:desc"
+
+        if not isinstance(sort_by, list):
+            sort_by = [sort_by]
+
+        # TODO: support also experiment_type
+        if len(sort_by) != 1:
+            raise ValueError("The fake service currently supports only sorting by start_datetime")
+
+        sortby_split = sort_by[0].split(":")
+        # TODO: support also experiment_type
+        if (
+                len(sortby_split) != 2
+                or sortby_split[0] != "start_datetime"
+                or (sortby_split[1] != "asc" and sortby_split[1] != "desc")
+        ):
+            raise ValueError(
+                "The fake service currently supports only sorting by start_datetime, which can be "
+                "either asc or desc"
+            )
+
+        df = df.sort_values(
+            ["start_time", "uuid"], ascending=[(sortby_split[1] == "asc"), True]
+        )
+
+        df = df.iloc[:limit]
+        result = {"experiments": df.replace({np.nan: None}).to_dict("records")}
```

Comment:
Why not call `serialize`

Reply:
`serialize` expects a dataframe and dumps it as-as; here we need to return a dictionary containing the dataframe's dict as a specific value. We can't return `serialize` because of this, and we also can't serialize `df` and add to `result` since in `result` we expect a `dict` before calling `json.dumps` in the next line.